src/main/java/intellistart/interviewplanning/model/slot/SlotRepository.kt

```diff
@@ -0,0 +1,82 @@
+package intellistart.interviewplanning.model.slot
+
+import intellistart.interviewplanning.model.user.Candidate
+import intellistart.interviewplanning.model.user.Interviewer
+import intellistart.interviewplanning.model.user.User
+import org.bson.types.ObjectId
+import org.springframework.data.mongodb.core.FindAndModifyOptions
+import org.springframework.data.mongodb.core.MongoTemplate
+import org.springframework.data.mongodb.core.aggregation.Aggregation
+import org.springframework.data.mongodb.core.query.Criteria
+import org.springframework.data.mongodb.core.query.Query
+import org.springframework.data.mongodb.core.query.Update
+import org.springframework.stereotype.Repository
+import java.time.LocalDate
+
+@Repository
+class SlotRepository(private val mongoTemplate: MongoTemplate) {
+
+    fun findByEmail(email: String): List<Slot> {
+        val aggregation = Aggregation.newAggregation(
+            Aggregation.match(Criteria.where("email").`is`(email)),
+            Aggregation.unwind("slots"),
+            Aggregation.replaceRoot("slots")
+        )
+
+        val results = mongoTemplate.aggregate(aggregation, User.COLLECTION_NAME, Slot::class.java)
+        return results.mappedResults
+    }
+
+    fun findById(id: ObjectId): Slot? {
```

Comment:
As i understand you , seams it can be simplified
```suggestion
   fun findById(id: ObjectId): Slot? {
    val query = Query(Criteria.where("slots._id").`is`(id))
    val user = mongoTemplate.findOne(query, User::class.java)
    return user?.slots?.find { it.id == id }
}
```

Reply:
it will not be correct because not all user child's have the slots field